spec/unit/provider/user/aix_spec.rb

```diff
@@ -84,6 +96,41 @@
     end
   end
 
+  describe '#gid=' do
+    let(:value) { 'new_pgrp' }
+
+    let(:old_pgrp) { 'old_pgrp' }
+    let(:cur_groups) { 'system,adm' }
+    before(:each) do
+      provider.stubs(:gid).returns(old_pgrp)
+      provider.stubs(:groups).returns(cur_groups)
+      provider.stubs(:set)
+    end
+
+    it 'raises a Puppet::Error if it fails to set the groups property' do
+      provider.stubs(:set)
+        .with(:groups, cur_groups)
+        .raises(Puppet::ExecutionFailure, 'failed to reset the groups!')
+
+      expect { provider.gid = value }.to raise_error do |error|
+        expect(error).to be_a(Puppet::Error)
+
+        expect(error.message).to match('groups')
+        expect(error.message).to match(cur_groups)
+        expect(error.message).to match(old_pgrp)
+        expect(error.message).to match(value)
+      end
+    end
+
+    it 'sets the primary group and resets the groups property' do
```

Comment:
On the same vein as the other comment: we probably don't need this as a unit test. Acceptance tests will cover this scenario and this looks to only be testing implementation, rather than functionality.